packages/medusa/src/services/system-tax.ts

```diff
@@ -9,8 +9,8 @@ import { ProviderTaxLine } from "../types/tax-service"
 class SystemTaxService extends AbstractTaxService {
   static identifier = "system"
 
-  constructor() {
-    super({})
+  constructor(_) {
+    super(_)
```

Comment:
@adrien2p can you confirm if it's ok to make this change? I needed to add the constructor to the `AbstractTaxService` for documentation purposes but the original constructor implementation caused a TypeScript error related to `{}` not matching type `MedusaContainer`.

Reply:
what I usually do is the following
```ts
class SystemTaxService extends AbstractTaxService {
  static identifier = "system"

  constructor(...args: any[]) {
    super(...arguments)
  }
```

Why? Because here what we are saying is that the SystemTaxService has a defined public constructor that can take any arguments and we pass all the given arguments to super (the base class). Now, maybe we would like to define specifically what arguments this class can take if someone wanted to extend it or just to specify what it is expected.

Reply:
I think it should be this right?

```ts
constructor(...args: any[]) {
    super(args)
  }
```

`args` instead of `arguments`.

I still get an error for incompatible types. For reference, this is how I've written the constructor:

https://github.com/medusajs/medusa/blob/f019d4e63b9c87728817cb0a4844958cb5518ba6/packages/medusa/src/interfaces/tax-service.ts#L237-L242

I based this implementation of both the constructor in the abstract class and in the system tax service on the implementation for the [system payment provider](https://github.com/medusajs/medusa/blob/develop/packages/medusa/src/services/system-payment-provider.ts).

Reply:
you need to spread the args. `arguments` is a special js keywords that hold all the arguments passed to a function. args here are the same as arguments. The difference is that if you always want to pass all the args provided by to the constructor `arguments` will always works

Reply:
I know, but I get this TS error when I use your codeblock:

```
A spread argument must either have a tuple type or be passed to a rest parameter.ts(2556)
```

Reply:
you can add ts-ignore

Reply:
Got it, I've added that. Thanks! 🙏